### PR TITLE
plugins:foreman_cockpit - Release 1.0.2 (RPM)

### PR DESCRIPTION
**Pull request by dLobatog**

### CODE REVIEW

**Comment by dLobatog** [comment]
Maybe it's better to hold off on this one and wait for a 1.0.2 that allows https cockpit content, this release will not work on https installations.

File: rubygem-foreman_cockpit/rubygem-foreman_cockpit.spec [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+%endif
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildRequires: %{?scl_prefix_ruby}rubygems
+
+BuildArch: noarch
+
+Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
+Provides: foreman-plugin-cockpit
+Provides: foreman-cockpit
+%{?scl:Obsoletes: ruby193-rubygem-%{gem_name}}
```
Comment (domcleal) [inline rubygem-foreman_cockpit/rubygem-foreman_cockpit.spec]: No need for this, the package never existed.

File: rubygem-foreman_cockpit/rubygem-foreman_cockpit.spec [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+%define rubyabi 1.9.1
+%global foreman_dir /usr/share/foreman
+%global foreman_bundlerd_dir %{foreman_dir}/bundler.d
+%global foreman_pluginconf_dir %{foreman_dir}/config/settings.plugins.d
+
+Summary:    Integration of Cockpit in Foreman
+Name:       %{?scl_prefix}rubygem-%{gem_name}
+Version:    1.0.1
+Release:    1%{?dist}
```
Comment (domcleal) [inline rubygem-foreman_cockpit/rubygem-foreman_cockpit.spec]: Please add %{foremandist} in here, so `1%{foremandist}%{dist}`

File: rubygem-foreman_cockpit/rubygem-foreman_cockpit.spec [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+Group:      Applications/System
+License:    GPLv3
+URL:        http://github.com/theforeman/foreman_cockpit
+Source0:    http://rubygems.org/downloads/%{gem_name}-%{version}.gem
+
+Requires:   foreman >= 1.6.0
```
Comment (domcleal) [inline rubygem-foreman_cockpit/rubygem-foreman_cockpit.spec]: 1.7.0 according to engine.rb

File: rubygem-foreman_cockpit/rubygem-foreman_cockpit.spec [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+Provides: foreman-cockpit
+%{?scl:Obsoletes: ruby193-rubygem-%{gem_name}}
+
+%description
+This plugin enables integration with Cockpit in Foreman.
+
+%package doc
+BuildArch:  noarch
+Requires:   %{?scl_prefix}%{pkg_name} = %{version}-%{release}
+%{?scl:Obsoletes: ruby193-rubygem-%{gem_name}-doc}
```
Comment (domcleal) [inline rubygem-foreman_cockpit/rubygem-foreman_cockpit.spec]: ditto

File: rubygem-foreman_cockpit/rubygem-foreman_cockpit.spec [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+%{?scl:%scl_package rubygem-%{gem_name}}
+%{!?scl:%global pkg_name %{name}}
+
+%global gem_name foreman_cockpit
+
+%define rubyabi 1.9.1
+%global foreman_dir /usr/share/foreman
+%global foreman_bundlerd_dir %{foreman_dir}/bundler.d
+%global foreman_pluginconf_dir %{foreman_dir}/config/settings.plugins.d
```
Comment (domcleal) [inline rubygem-foreman_cockpit/rubygem-foreman_cockpit.spec]: unused

**Comment by domcleal** [comment]
A couple of other points:

1. You need asset compilation since the plugin ships a Javascript file.  Take a look at foreman_setup or similar, you'll need foreman-plugin as a BR, then use `%foreman_precompile_plugin -s` and to update the %files list.  It also means you can get rid of the %foreman definitions at the top, and the custom bundler.d `cat` line, since the macros we ship can do it all for you.
1. Missing rel-eng/comps changes, see https://github.com/theforeman/foreman-packaging/tree/rpm/develop#howto-create-a-new-core-package-or-dependency

File: rubygem-foreman_cockpit/rubygem-foreman_cockpit.spec [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+gem '%{gem_name}'
+GEMFILE
+
+%files
+%dir %{gem_instdir}
+%{gem_instdir}/app
+%{gem_instdir}/config
+%{gem_instdir}/lib
+%{gem_instdir}/locale
+%exclude %{gem_cache}
```
Comment (domcleal) [inline rubygem-foreman_cockpit/rubygem-foreman_cockpit.spec]: dupe with line 98

**Comment by dLobatog** [comment]
@domcleal Updated, thanks for the comments

**Comment by dLobatog** [comment]
[test] - I didn't push the gem before submitting the PR

**Comment by mmoll** [comment]
[test] again

**Comment by domcleal** [comment]
Check the build log via Jenkins:

<pre>
+ cp -a ./opt/theforeman/tfm/root/usr/share/gems/cache ./opt/theforeman/tfm/root/usr/share/gems/doc ./opt/theforeman/tfm/root/usr/share/gems/gems ./opt/theforeman/tfm/root/usr/share/gems/specifications /builddir/build/BUILDROOT/tfm-rubygem-foreman_cockpit-1.0.2-1.fm1_11.git.0.f8eb9e8.git.0.f8eb9e8.el6.noarch/opt/theforeman/tfm/root/usr/share/gems/
+ %foreman_bundlerd_file
/var/tmp/rpm-tmp.VmPxYZ: line 40: fg: no job control
</pre>

Suggests you're missing the `foreman-plugin` BR, as it's not able to expand that macro.

File: rubygem-foreman_cockpit/rubygem-foreman_cockpit.spec [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+# http://docs.fedoraproject.org/en-US/Fedora_Contributor_Documentation
+# /1/html/Software_Collections_Guide/index.html
+
+%{?scl:%scl_package rubygem-%{gem_name}}
+%{!?scl:%global pkg_name %{name}}
+
+%global gem_name foreman_cockpit
+
+%define rubyabi 1.9.1
+%global foreman_dir /usr/share/foreman
```
Comment (domcleal) [inline rubygem-foreman_cockpit/rubygem-foreman_cockpit.spec]: unused, defined by foreman-plugin

**Comment by domcleal** [comment]
Thanks @dLobatog, building now.

File: rubygem-foreman_cockpit/rubygem-foreman_cockpit.spec (hunk 1 above)
Comment (domcleal) [inline rubygem-foreman_cockpit/rubygem-foreman_cockpit.spec]: I've restored this because it was introduced in #871 for 1.8/1.9.

**Pull request merged**